src/map/status_effect_container.cpp

```diff
@@ -488,6 +488,11 @@ bool CStatusEffectContainer::AddStatusEffect(CStatusEffect* PStatusEffect, bool
         return false;
     }
 
+    if (m_POwner->PAI->GetController() == NULL) {
```

Comment:
This is where Controller was null when zoning into jail, which caused the newly added onEffectLose side effect of hundred fists to crash.
This should not happen anymore with the noop controller, but just to avoid a future crash, good to have the check here.